### PR TITLE
python3-numpy: update to 2.1.0.

### PR DESCRIPTION
**Pull request by tornaria**
After the update, there's a conflict with flintlib headers which causes
trouble, for instance when building sagemath.

A patch to flintlib is included here (from https://github.com/flintlib/flint/pull/2027, already merged upstream).

@ahesford : I'm testing sagemath locally and here as well.

#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

**Comment by tornaria** [comment]
sagemath needs some patching for numpy 2.1, nothing too serious...

I will look into it and report here when it's ready.

File: srcpkgs/sagemath/patches/numpy_2.1b.patch [hunk 1]
```diff
@@ -0,0 +1,223 @@
+diff --git a/src/sage/combinat/fully_packed_loop.py b/src/sage/combinat/fully_packed_loop.py
+index b3ebd206cd2..d61242928dc 100644
+--- a/src/sage/combinat/fully_packed_loop.py
++++ b/src/sage/combinat/fully_packed_loop.py
+@@ -68,7 +68,7 @@ def _make_color_list(n, colors=None, color_map=None, randomize=False):
+ 
+         sage: import numpy as np
+         sage: if int(np.version.short_version[0]) > 1:
+-        ....:     np.set_printoptions(legacy="1.25")
++        ....:     np.set_printoptions(legacy="1.25") and None
```
Comment (tornaria) [inline srcpkgs/sagemath/patches/numpy_2.1b.patch:10]: @ahesford this works but I'm not convinced, what do you think?

Point is,
- in numpy 2.0, `np.set_printoptions()` returns `None` which sagemath does not print.
- in numpy 2.1, it returns some object which sagemath prints. I hoped semicolon would do the trick, but it doesn't.

Comment (tornaria) [reply srcpkgs/sagemath/patches/numpy_2.1b.patch:10]: Maybe this is the reasonable way:
```suggestion
         sage: if int(np.version.short_version[0]) > 1:
-        ....:     np.set_printoptions(legacy="1.25")
+        ....:     _token = np.set_printoptions(legacy="1.25")
```

**Comment by tornaria** [comment]
I added here the scipy update which seems to be working just fine.

**Comment by ahesford** [comment]
Thanks!

**Pull request merged**